### PR TITLE
Glm log log

### PR DESCRIPTION
**Pull request by gmcmacran**
This pull requests adds the loglog link to the glm function for the binomial family. 

Changes are
 #1 Create class LogLog and loglog
#2  Edit binimial family to use #1's class
#3 Add Loglog to tests used for other links

In addition to automated tests, results were validated against R's extendedFamily package and example from Generalized Linear Models and Extensions book.

This is my first contribution to this repo. I attempted to follow https://www.statsmodels.org/stable/dev/index.html. Feedback/corrections are welcome.

- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/genmod/families/tests/test_link.py [hunk 1]
```diff
@@ -31,6 +32,8 @@ def get_domainvalue(link):
     z = -np.log(np.random.uniform(0, 1))
     if isinstance(link, links.CLogLog):  # prone to overflow
         z = min(z, 3)
+    elif isinstance(link, links.CLogLog):
```
Comment (josef-pkt) [inline statsmodels/genmod/families/tests/test_link.py:35]: you have CLogLog here, and not LogLog

**Comment by josef-pkt** [comment]
Thanks, looks good.

some whitespace issues, you need to strip trailing whitespace.
I need to check before merging how much automatic test coverage we have for this.

**Comment by josef-pkt** [comment]
can you add `inverse_deriv2`?    second derivative of inverse link
then we wouldn't have to compute it indirectly.

AFAIR, the plan is to have those also explicitly now.
I only added them to CDFLinks so far

**Comment by josef-pkt** [comment]
Thanks, looks good now, unit tests are green

2 remaining pep-8 style check complaints
https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=3498&view=logs&j=5999b18d-e094-51e5-d97a-d1e11af2567f&t=d028493a-5398-5672-bc8e-00ba508ef3fb
blank after (, and one line too long

Do you know how to squash commits in an interactive rebase?

**Comment by josef-pkt** [comment]
Great, good to get this missing link added. Thanks again.

PR looks done

maybe squashing commits, 
and I want to go over unit tests because I don't remember how they are structured for this.
otherwise looks ready to merge

**Comment by gmcmacran** [comment]
> 
> 
> Thanks, looks good now, unit tests are green
> 
> 2 remaining pep-8 style check complaints
> https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=3498&view=logs&j=5999b18d-e094-51e5-d97a-d1e11af2567f&t=d028493a-5398-5672-bc8e-00ba508ef3fb
> blank after (, and one line too long
> 
> Do you know how to squash commits in an interactive rebase?

Thanks for the help. I did some reading around squashing commits and changing the base. I think the safest thing is for you to do it if possible. I have stuck to the basics of version control. Is this possible? If not, I can make a practice repo and figure it out.

**Comment by gmcmacran** [comment]
Been a few days. Checking in. Please advise if anything else needs to be done for this to be merged. Thanks.

**Comment by josef-pkt** [comment]
Sorry, I was stuck in beta regression and didn't want to switch topics and branches before merging that.

**Comment by josef-pkt** [comment]
merged after rebase in #7562

**Comment by josef-pkt** [comment]
Thanks @gmcmacran

**Pull request closed without merging**